### PR TITLE
PA-17 - Authentication: sign up, sign in

### PR DESCRIPTION
**Pull request by sxiixii**
## What does this PR do?

Add sign up and sign in page. 

## How it can be tested?

Go to `backend` folder

Install requirements
`pip install -r requirements`

Apply migrations
'./manage.py migrate'

Run server
`./manage.py runserver`

## PR Checklist
Please check if your PR fulfills the following requirements:
 - [X] I tested my code

### CODE REVIEW

**Review by TondeMoon — APPROVED** [review]
Seems all nice to me, but some refactoring commits are named not exactly according to instructions.

**Comment by sxiixii** [comment]
> Seems all nice to me, but some refactoring commits are named not exactly according to instructions.

You're right. This is my fault.

File: frontend/src/App.test.tsx [hunk 1]
```diff
@@ -1,57 +1,15 @@
...
-import App from './App'
-import Counter from './components/counter/Counter'
-import userEvent from '@testing-library/user-event'
-import {createStore, Store} from 'redux'
+import {render, screen} from '@testing-library/react'
 import { Provider } from 'react-redux'
-import counterReducer from './reducers/counterReducer'
-import { CounterState } from './actions'
+import '@testing-library/jest-dom'
+// import '@types/jest'
```
Comment (ivan-gerasin) [inline frontend/src/App.test.tsx]: Remove commented code

File: frontend/src/App.tsx [hunk 2]
```diff
@@ -1,22 +1,35 @@
-import React, {useEffect} from 'react'
-import { createStore } from 'redux'
-import { Provider } from 'react-redux'
-import Counter from './components/counter/Counter'
-import counterReducer from '../src/reducers/counterReducer'
-const store = createStore(counterReducer)
+import React from 'react'
+import { useSelector } from 'react-redux'
```
Comment (ivan-gerasin) [inline frontend/src/App.tsx]: Please fix following linter warnings

File: frontend/src/App.tsx [hunk 3]
```diff
@@ -1,22 +1,35 @@
...
+import {AppStateType} from './index'
+import SignUpSide from '../src/components/authentication/SignUp'
+import Welcome from './components/personal/PersonalPage'
+import SignInSide from './components/authentication/SignIn'
 
 const App: React.FC = () => {
-  useEffect(() => {
-    fetch(`${process.env.REACT_APP_DJANGO_SERVER}/api/v1/hello/`)
-  })
+  // const isAuth = useSelector<AppStateType >(state => state.loggedIn)
```
Comment (ivan-gerasin) [inline frontend/src/App.tsx]: unused code

File: frontend/src/actions/authActions.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+export type fetchUserAction = Action<AuthActionType.SET_USER, IUser>
+export type logoutUserAction = Action<AuthActionType.LOG_OUT>
+
+export type AuthActions = fetchUserAction | logoutUserAction
+
+export const setUserAction = actionCreator<AuthActionType.SET_USER, IUser>(AuthActionType.SET_USER)
+
+// TODO: localStorage.clear()
```
Comment (ivan-gerasin) [inline frontend/src/actions/authActions.ts]: When we want to implement this TODO? Please, fix this, or file a new ticket for this and left a reference in comment.

File: frontend/src/api/authRequest.test.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+jest.mock('./HttpClient')
+
+describe('use of AuthApiClient', () => {
+  beforeEach(() => {
+    HttpClient.mockClear()
+  })
+  
+  it('We can check if authClient called the class constructor', () => {
+    const authClient = new AuthApiClient()
```
Comment (ivan-gerasin) [inline frontend/src/api/authRequest.test.tsx]: please fix linter warnings

**Review by ivan-gerasin — APPROVED** [review]
Nice! I have only few minor comments on frontend, so you can merge PR by yourself after it get fixed.

File: frontend/src/actions/authActions.ts (hunk 4 above)
Comment (TondeMoon) [reply frontend/src/actions/authActions.ts]: added PA-55 for Logout to make Logout and fix LocalStorage.clear()

**Pull request merged**